pyrit/score/general_scorer.py

```
@@ -0,0 +1,103 @@
+    Params:
+        chat_target (PromptChatTarget): The chat target to use for scoring.
+        system_prompt (str): The system-level prompt that guides the behavior of the target LLM. Defaults to None.
+        prompt_fstring_format (str): The format string for the prompt. Defaults to None.
+        scorer_type (str): The type of scorer (e.g., "true_false", "float_scale"). defaults to float_scale.
+        category (list): A list of categories for the score.
+        labels (list): A list of labels for the score.
+        min_value (int): The minimum value for float scale scoring.
+        max_value (int): The maximum value for float scale scoring.
+        output_keys (dict): A dictionary of output keys for the score response.
+    """
+
+    def __init__(
+        self,
+        chat_target: PromptChatTarget,
+        system_prompt: Optional[str] = None,
+        prompt_fstring_format: str = None,
+        scorer_type: str = "float_scale",
+        category: list = None,
+        labels: list = None,
+        min_value: Optional[int] = 0,
+        max_value: Optional[int] = 100,
+        output_keys: Optional[dict] = None,
+    ) -> None:
+        self._prompt_target = chat_target
+        self._system_prompt = system_prompt
+        self.prompt_fstring_format = prompt_fstring_format
+        self.scorer_type = scorer_type
+        self._score_category = category
+        self.labels = labels
+        self._min_value = min_value
+        self._max_value = max_value
+        self.prompt_fstring_format = prompt_fstring_format
+
+        # Default output keys
+        default_output_keys = {
+            "score_value": "score_value",
+            "rationale": "rationale",
+            "metadata": "metadata",
+            "description": "description",
+        }
+
+        # Merge default keys with provided keys (if any)
+        self._output_keys = {**default_output_keys, **(output_keys or {})}
```

nit: if a user supplies both type true_false and a scale we should throw an exception.